Feat: Post 저장 기능 구현

Shsin9797 opened the pull request:
## 🔎 작업 내용

> Post save 기능을 구현했습니다. 
 400,500 처리는 할예정 입니니다. 

## 💬 리뷰 요구사항 (선택)

1. response 값에 어떤걸 넣어주는게 포멀 한가요? 

- 게시글의 경우, 게시글 id 와 게시글 제목 정도만 반환해도 되나요? 아니면 게시글 내용도 반환하는게 좋나요 ? 
- 생성일자 안넣어도 되나요? 

2. document 보고+ gpt 내용 참고해서  ResponseEntity 에 새로생성된 게시글의 url 을 헤더에 넣어주는 방식으로 구현했는데, url fomat  생성하는 부분이 코드변경에 취약할 것 같아 보입니다. 이부분을 개선할 방법이 있나요? 아니면 이렇게 코드 작성하는게 잘못되었는지 궁금합니다. 

## 📸 이미지 첨부 (선택)
1. 현재는 게시글 id랑 title 정도만 반환하고 있습니다. 
![image](https://github.com/user-attachments/assets/b96178c2-f45a-4571-9b35-21623bf3e05e)

2. 이부분에서 v1, v2 가 추가된다거나 하는 url 이 변경된경우 , 일일이 찾아가서 수정하기 힘들어보입니다.  
![image](https://github.com/user-attachments/assets/f899e12e-5fc8-48dc-8aba-166f4296e95b)



## ➕ 이슈 링크

- [Feat:Post #6]https://github.com/kakao-tech-campus-2nd-step3/Team29_BE/issues/6

Shsin9797 committed:
Feat: Post 저장 기능 구현

Shsin9797 committed:
Feat: Post Id로 Post 1개 찾기 기능 구현

hynseoj commented on src/main/java/notai/post/presentation/PostController.java:
```diff
@@ -1,4 +1,44 @@
+            @RequestBody PostSaveRequest postSaveRequest
+    ) {
+        PostSaveResult postSaveResult;
+        postSaveResult = postService.savePost(postSaveRequest);
```
여기는 선언과 할당을 한 줄로 작성하는 게 보기 더 편할 것 같습니다!
`PostSaveResult result = postService.savePost(postSaveRequest);`

hynseoj replied:
계층별로 응답을 result 랑 response 로 잘 분리해주셨네요!
요청도 request 랑 command 로 분리해주시면 더 좋을 것 같아요!!

Shsin9797 replied:
넵 리뷰 감사합니다!! 말씀해주신부분 반영하겠습니다!

Shsin9797 replied:
> 저는 생성의 경우에는 지금 작성하신 것처럼 created 에 uri 를 반환하도록 한다면 response body 는 없어도 된다고 생각하긴합니다!

생각해보니 그런거같네요! 감사합니다! 

생성시에는 반환값을 uri만 헤더에 담아 반환하는걸로 수정했습니다. 

그런데 이때 PostSaveResponse 가 필요없는것 같아서 삭제했는데 괜찮은지 궁금합니다! (잘몰라가지구 ..) 

몬가 통일성이 없어서 괜찮으려나 싶기도 한데 그런건 상관없이 꼭필요한 코드만 짜는게 더 나은 방향일것 같기도 합니다!

Shsin9797 replied:
요청을 request 와 command로 분리 완료했습니다!

hynseoj replied:
> 그런데 이때 PostSaveResponse 가 필요없는것 같아서 삭제했는데 괜찮은지 궁금합니다! (잘몰라가지구 ..)
몬가 통일성이 없어서 괜찮으려나 싶기도 한데 그런건 상관없이 꼭필요한 코드만 짜는게 더 나은 방향일것 같기도 합니다!

헉 너무 늦게봤네요...!!!!!! 괜찮을 거 같아요!! 👍🥰😊

hynseoj replied:
> 그런데 이때 PostSaveResponse 가 필요없는것 같아서 삭제했는데 괜찮은지 궁금합니다! (잘몰라가지구 ..)
몬가 통일성이 없어서 괜찮으려나 싶기도 한데 그런건 상관없이 꼭필요한 코드만 짜는게 더 나은 방향일것 같기도 합니다!

헉 너무 늦게봤네요...!!!!!! 괜찮을 거 같아요!! 👍🥰😊

hynseoj replied:
> 그런데 이때 PostSaveResponse 가 필요없는것 같아서 삭제했는데 괜찮은지 궁금합니다! (잘몰라가지구 ..)
몬가 통일성이 없어서 괜찮으려나 싶기도 한데 그런건 상관없이 꼭필요한 코드만 짜는게 더 나은 방향일것 같기도 합니다!

헉 너무 늦게봤네요...!!!!!! 괜찮을 거 같아요!! 👍🥰😊

hynseoj reviewed (changes requested):
고생하셨습니다~ 전체적으로 잘 구현해주신 것 같아요!
요청만 request 랑 command 로 분리해주시면 approve 하겠습니다!

> response 값에 어떤걸 넣어주는게 포멀 한가요?
게시글의 경우, 게시글 id 와 게시글 제목 정도만 반환해도 되나요? 아니면 게시글 내용도 반환하는게 좋나요 ?
생성일자 안넣어도 되나요?

저는 생성의 경우에는 지금 작성하신 것처럼 created 에 uri 를 반환하도록 한다면 response body 는 없어도 된다고 생각하긴합니다!

> document 보고+ gpt 내용 참고해서 ResponseEntity 에 새로생성된 게시글의 url 을 헤더에 넣어주는 방식으로 구현했는데, url fomat 생성하는 부분이 코드변경에 취약할 것 같아 보입니다. 이부분을 개선할 방법이 있나요? 아니면 이렇게 코드 작성하는게 잘못되었는지 궁금합니다.

음.. 상수로 따로 빼는 정도가 있을 것 같아요 👀

yugyeom-ghim commented on src/main/java/notai/post/query/PostQueryRepository.java:
코딩 스타일의 차이일수도있는데

queryRepository는 queryDSL을 사용할때 작성하고 JpaRepository를 사용할때는 그냥 PostRepository로 하는것도 괜찮을 것 같습니다

Shsin9797 replied:
아하 그렇군요, QueryRepository 나 QueryService 는 조회관련 쿼리를 모아두는 곳으로 이해하고 있었네요.. 

제가 구현할부분은 사실 아직까진 크게 복잡한 조회 쿼리가 필요하진 않아보이기도 하고, 조회하는 부분을 따로 떼어둘 필요는 없어 보이네요 

유겸님께서 말씀해주신거처럼 QueryDSL 등의 쿼리를 모아 작성하는 것도 좋은 방법인것 같습니다. 

수정했습니다. 감사합니다!!

yugyeom-ghim reviewed (approved):
고생하셨습니다!

Shsin9797 committed:
Style: 코드리뷰 반영 - 변수선언과 변수 할당을 한줄에 하도록 변경

Shsin9797 committed:
Refactor: 코드리뷰 반영 : Post 조회 수행 위치 변경
- JPA 사용하는 단순 조회 쿼리를 PostService 와 PostRepository 에서 관리하도록 변경 (기존에는 PostQueryService 와 PostQueryRepository에서 관리했음)

Shsin9797 committed:
Fix: 코드리뷰 반영 : Post 조회시 ResponseBody 삭제
-uri 반환중이라 굳이 불필요해서 삭제함

Shsin9797 committed:
Fix: 코드리뷰 반영 : Post 저장 기능에서 Request와 Command 를 분리

hynseoj reviewed (approved):
고생하셨습니다!!
혹시 pr 보내는 데를 Weekly8 으로 변경해주실 수 있으실까요 🥹
